paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/FlowControllerConfigurationHandler.kt

```diff
@@ -1,77 +1,90 @@
-import kotlinx.coroutines.isActive
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CompletableJob
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
 import java.security.InvalidParameterException
 import javax.inject.Inject
 import javax.inject.Singleton
-import kotlin.coroutines.CoroutineContext
 
 @Singleton
 internal class FlowControllerConfigurationHandler @Inject constructor(
     private val paymentSheetLoader: PaymentSheetLoader,
-    @UIContext private val uiContext: CoroutineContext,
     private val eventReporter: EventReporter,
     private val viewModel: FlowControllerViewModel,
     private val paymentSelectionUpdater: PaymentSelectionUpdater,
 ) {
 
-    suspend fun configure(
+    private var job: CompletableJob? = null
+
+    fun configure(
+        scope: CoroutineScope,
         initializationMode: PaymentSheet.InitializationMode,
         configuration: PaymentSheet.Configuration?,
         callback: PaymentSheet.FlowController.ConfigCallback,
+    ) {
+        job?.cancel()
+
+        job = Job().apply {
+            invokeOnCompletion { error ->
+                if (error !is CancellationException) {
+                    callback.onConfigured(
+                        success = error == null,
+                        error = error,
+                    )
+                    job = null
+                }
+            }
+        }
+
+        scope.launch(job!!) {
+            configureInternal(
+                initializationMode = initializationMode,
+                configuration = configuration,
+            )
+        }
+    }
+
+    private suspend fun configureInternal(
+        initializationMode: PaymentSheet.InitializationMode,
+        configuration: PaymentSheet.Configuration?,
     ) {
         try {
             initializationMode.validate()
             configuration?.validate()
         } catch (e: InvalidParameterException) {
-            callback.onConfigured(false, e)
+            job?.completeExceptionally(e)
             return
         }
 
         val configureRequest = ConfigureRequest(initializationMode, configuration)
         val canSkip = viewModel.previousConfigureRequest == configureRequest
 
         if (canSkip) {
-            callback.onConfigured(true, null)
+            job?.complete()
             return
         }
 
-        val result = paymentSheetLoader.load(initializationMode, configuration)
-
-        if (currentCoroutineContext().isActive) {
-            viewModel.previousConfigureRequest = configureRequest
-            dispatchResult(result, callback)
-        } else {
-            callback.onConfigured(false, null)
-        }
-    }
-
-    private suspend fun dispatchResult(
-        result: PaymentSheetLoader.Result,
-        callback: PaymentSheet.FlowController.ConfigCallback,
-    ) = withContext(uiContext) {
```

Comment:
I think this means we're no longer dispatching the result to the main thread. I'd consider this a (breaking) behavior change, right?